sdk/identity/azure-identity/src/chained_token_credential.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/identity/chained_token_credential.hpp"
+
+#include <utility>
+
+using namespace Azure::Identity;
+
+ChainedTokenCredential::ChainedTokenCredential(ChainedTokenCredential::Sources sources)
+    : m_sources(std::move(sources))
+{
+}
+
+ChainedTokenCredential::~ChainedTokenCredential() = default;
+
+Azure::Core::Credentials::AccessToken ChainedTokenCredential::GetToken(
+    Azure::Core::Credentials::TokenRequestContext const& tokenRequestContext,
+    Azure::Core::Context const& context) const
+{
+  for (const auto& source : m_sources)
+  {
+    try
+    {
+      return source->GetToken(tokenRequestContext, context);
+    }
+    catch (const Azure::Core::Credentials::AuthenticationException&)
+    {
+      // Do nothing, next source will be tried.
```

Comment:
Do we want to at least add a log entry about this?

Do we log anything about getting a token?   I think any authPolicy creates a pipeline to get a token which has a logging policy on it right?   So, it would be useful to see a log entry from the chain credential before logging each credential intent one after the other?